migration/promote.sh

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
+
+# Script to promote the upgrade and migration for releases.
+
+set -euo pipefail
+
+HERE="$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")"
+ROOT="$(dirname "${HERE}")"
+
+source "${ROOT}/scripts/common.sh"
+
+usage() {
+  echo -n "usage:
+  - $0 $(esc.ylw "<release-series>") - promotes the upgrade and migration for the given release series in the form $(esc.ylw "vX.Y")
+  "
+}
+
+main() {
+  local series="${1:-}"
+
+  [[ -n "${series}" ]] || log.fatal "missing required argument $(esc.ylw "<release-series>"), $(usage)"
+
+  ver.parse next "${series}"
+
+  local target_version="v${ver["next-major"]}.${ver["next-minor"]}"
+  local target_migration="${ROOT}/migration/${target_version}"
+
+  if [[ -d "${target_migration}" ]]; then
+    if log.continue "migration $(esc.ylw "${target_version}") already exists, do you want to recreate it?"; then
+      log.warn "removing migration $(esc.ylw "${target_version}")"
+      rm -r "${target_migration}"
+    else
+      log.warn "skipping"
```

Comment:
What is it meant to skip?

Should it maybe `exit` here?

If not, is this expected?:

``` console
$ bash ./migration/promote.sh v0.43 
[ck8s/promote.sh] migration v0.43 already exists, do you want to recreate it? - [y/N]: 
[ck8s/promote.sh] skipping
Error: bad file '/tmp/temp2072476512': yaml: did not find expected comment or line break
```

I guess `yq4` has trouble with the missing front matter section in existing migration `README`s

Reply:
It is meant to skip the full recreation of the migration folder, so a complete wipe and copy.
It will still replace references and update text, but maybe it should skip that and just continue to prune. :thinking: 

And yes I assume it is the missing front matter, I would not run it on v0.43 and older which is using the old template.

Reply:
Now that error cannot happen due to old readmes, pruning will still kick in but that is separate. :slightly_smiling_face: